src/pru-motion-queue.cc

```diff
@@ -82,54 +82,18 @@ static void DumpMotionSegment(volatile const struct MotionSegment *e,
 }
 #endif
 
-// Store the required informations needed to backtrack the absolute position.
-struct HistorySegment {
-  HistorySegment() : fractions(), cumulative_loops(), direction_bits(0) {}
-  uint32_t fractions[MOTION_MOTOR_COUNT];
-  uint32_t cumulative_loops[MOTION_MOTOR_COUNT];
-  uint8_t direction_bits;
-};
-
-void PRUMotionQueue::RegisterHistorySegment(const MotionSegment &element) {
-  const uint64_t max_fraction = 0xFFFFFFFF;
-  const unsigned int last_insert_index = (queue_pos_ - 1) % QUEUE_LEN;
-  const struct HistorySegment &previous
-    = shadow_queue_[(last_insert_index - 1) % QUEUE_LEN];
-
-  const uint64_t total_loops
-    = element.loops_accel + element.loops_travel + element.loops_decel;
-  const uint8_t direction_bits = element.direction_bits;
-
-  HistorySegment *new_slot = &shadow_queue_[last_insert_index];
-
-  // TODO: Motor operations already holds this information, we should wire it
-  // until here in order to avoid this redundant operation.
-  for (int i = 0; i < MOTION_MOTOR_COUNT; ++i) {
-    const uint64_t fraction = element.fractions[i];
-    new_slot->fractions[i] = fraction;
-    const int sign = (direction_bits >> i) & 1 ? -1 : 1;
-    uint64_t loops = fraction * total_loops;
-    loops /= max_fraction;
-    new_slot->cumulative_loops[i] = previous.cumulative_loops[i]
-                                    + sign * loops;
-  }
-
-  new_slot->direction_bits = element.direction_bits;
-}
-
-void PRUMotionQueue::GetMotorsLoops(MotorsRegister *absolute_pos_loops) {
-  const uint64_t max_fraction = 0xFFFFFFFF;
+void PRUMotionQueue::Status(uint32_t *executed_loops,
+                            unsigned int *buffer_size) {
```

Comment:
Given that it probably would be more work to come up with a good explanation about the meaning of the buffer_size that we can put in the comment, it would be better if we do this change now.